ldso/ldso/arc/resolve.S

```diff
@@ -12,30 +12,47 @@
 ; 	r10-r12 are already clobbered by PLTn, PLT0 thus neednot be saved
 
 .macro	SAVE_CALLER_SAVED
+#ifdef __ARC64_ARCH32__
+	push	r0
+	push	r1
+	push	r2
+	push	r3
+#else
 	push_s	r0
```

Comment:
Does it make sense to create some defines like PUSH_S/POP_S/ADD_S instead of using ifdef in each assembly macro?

Reply:
I'll second what Vladimir said. Please just use `PUSHR` and `POPR`. Something like these maybe:

[[arc32/libgloss] Make crt0 arc32 friendly](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/commit/457fe22f6b614e7dd755c198fda05b016e7de514)
[arc32: Make crt*.S arc32 friendly](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/commit/1db624110f6dc17264257e3820249c47d404ccb9)

Reply:
Done